src/3d/shaders/clipplane.shaderinc

```diff
@@ -0,0 +1,18 @@
+const int MAX_PLANE = 8;
+uniform int max_plane_real;
+
+uniform vec4 clipPlane[MAX_PLANE];
+
+out float gl_ClipDistance[MAX_PLANE];
+
+void setClipDistance(vec3 worldPos)
+{
+  for (int i = 0; i < max_plane_real; ++i)
+    {
+      // gl_ClipDistance is the signed distance from vertex to plane
+      // if the distance is positive, the vertex is kept
+      // if the distance is negative, the vertex is dropped
```

Comment:
This is not entirely correct. It's not that OpenGL would drop the vertex altogether - it will keep the calculated clip distance, and then for individual pixels of triangles it will interpolate the clip distance value and decide whether to clip the pixel or not (hence it works well even if a clipping plane splits a triangle - if vertices would be dropped, there would be badly rendered triangles).

Reply:
Dropped it.